lightning/src/ln/peer_handler.rs

```diff
@@ -33,22 +33,131 @@
+	fn get_and_clear_pending_msg_events(&self) -> Vec<MessageSendEvent> {
+		let mut res = Vec::new();
+		mem::swap(&mut res, &mut self.message_queue.lock().unwrap());
+		res
+	}
+}
+impl ChannelMessageHandler for DummyChannelHandler {
+	fn handle_open_channel(&self, their_node_id: &PublicKey, _their_features: InitFeatures, msg: &msgs::OpenChannel) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.temporary_channel_id);
+	}
+	fn handle_accept_channel(&self, their_node_id: &PublicKey, _their_features: InitFeatures, msg: &msgs::AcceptChannel) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.temporary_channel_id);
+	}
+	fn handle_funding_created(&self, their_node_id: &PublicKey, msg: &msgs::FundingCreated) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.temporary_channel_id);
+	}
+	fn handle_funding_signed(&self, their_node_id: &PublicKey, msg: &msgs::FundingSigned) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_funding_locked(&self, their_node_id: &PublicKey, msg: &msgs::FundingLocked) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_shutdown(&self, their_node_id: &PublicKey, _their_features: &InitFeatures, msg: &msgs::Shutdown) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_closing_signed(&self, their_node_id: &PublicKey, msg: &msgs::ClosingSigned) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_add_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateAddHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fulfill_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFulfillHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fail_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFailHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fail_malformed_htlc(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFailMalformedHTLC) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_commitment_signed(&self, their_node_id: &PublicKey, msg: &msgs::CommitmentSigned) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_revoke_and_ack(&self, their_node_id: &PublicKey, msg: &msgs::RevokeAndACK) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_update_fee(&self, their_node_id: &PublicKey, msg: &msgs::UpdateFee) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn handle_announcement_signatures(&self, their_node_id: &PublicKey, msg: &msgs::AnnouncementSignatures) {
+		DummyChannelHandler::push_error(self, their_node_id, msg.channel_id);
+	}
+	fn peer_disconnected(&self, _their_node_id: &PublicKey, _no_connection_possible: bool) {}
```

Comment:
I'd move all the empty methods together (`peer_(dis)connect` and `handle_error`) and perhaps add a comment as to why those three in particular don't need the `push_error` call